src/Go.Windows/GoBuildCommand.cs

```diff
@@ -0,0 +1,152 @@
+        /// Command menu group (command set GUID).
+        /// </summary>
+        public static readonly Guid CommandSet = new Guid("13629496-9f13-45fd-96d9-b5ba4ac9c086");
+
+        /// <summary>
+        /// VS Package that provides this command, not null.
+        /// </summary>
+        private readonly AsyncPackage package;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GoBuildCommand"/> class.
+        /// Adds our command handlers for menu (commands must exist in the command table file)
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        /// <param name="commandService">Command service to add command to, not null.</param>
+        private GoBuildCommand(AsyncPackage package, OleMenuCommandService commandService)
+        {
+            this.package = package ?? throw new ArgumentNullException(nameof(package));
+            commandService = commandService ?? throw new ArgumentNullException(nameof(commandService));
+
+            var menuCommandID = new CommandID(CommandSet, CommandId);
+            MenuCommand menuItem = new MenuCommand(this.BuildGoFile, menuCommandID);
+            commandService.AddCommand(menuItem);
+        }
+
+        private void BuildGoFile(object sender, EventArgs e)
+        {
+            // Show a message box to prove we were here
+            VsShellUtilities.ShowMessageBox(
+                this.package,
+                "Run go build",
+                "Go Build",
+                OLEMSGICON.OLEMSGICON_INFO,
+                OLEMSGBUTTON.OLEMSGBUTTON_OK,
+                OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
+
+            // Try to get document path
+            var dte = (ServiceProvider.GetServiceAsync(typeof(DTE))).Result as DTE;
+            var dteServiceProvider = new ServiceProvider((Microsoft.VisualStudio.OLE.Interop.IServiceProvider)dte);
+            var documentService = dteServiceProvider.GetService(typeof(Document)) as Document;
+            if (documentService != null)
+            {
+                var documentPath = documentService.Path;
+                Console.WriteLine(documentPath);
+            }
+
+            // Try to output something
+            var outputWindowPane = ServiceProvider.GetServiceAsync(typeof(IVsOutputWindowPane)).Result as IVsOutputWindowPane;
+            if (outputWindowPane != null)
+            {
+                outputWindowPane.Activate();
+                outputWindowPane.OutputStringThreadSafe("Trying to run go build command");
+            }
+            
+            var processInfo = new ProcessStartInfo("go.exe");
+            processInfo.Arguments = "build -v";
+            processInfo.RedirectStandardError = true;
+            processInfo.UseShellExecute = false;
+            System.Diagnostics.Process.Start(processInfo);
+        }
+
+        private string GetDocumentPath(ITextSnapshot textSnapshot)
+        {
+            ITextDocument textDoc;
```

Comment:
There's an API for this: https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.itextdocumentfactoryservice.trygettextdocument?view=visualstudiosdk-2019#Microsoft_VisualStudio_Text_ITextDocumentFactoryService_TryGetTextDocument_Microsoft_VisualStudio_Text_ITextBuffer_Microsoft_VisualStudio_Text_ITextDocument__

Reply:
Removed this method. Will experiment on that service.